e2e/imgbuild/imgbuild.go

```diff
@@ -1530,43 +1530,130 @@ echo 'export LEGACY_TEST_ENV=legacy-value' >> $SINGULARITY_ENVIRONMENT
-		"issue 4967":                      c.issue4967,                     // https://github.com/apptainer/singularity/issues/4967
-		"issue 4969":                      c.issue4969,                     // https://github.com/apptainer/singularity/issues/4969
-		"issue 5166":                      c.issue5166,                     // https://github.com/apptainer/singularity/issues/5166
-		"issue 5172":                      c.issue5172,                     // https://github.com/apptainer/singularity/issues/5172
-		"issue 5250":                      c.issue5250,                     // https://github.com/apptainer/singularity/issues/5250
-		"issue 5315":                      c.issue5315,                     // https://github.com/apptainer/singularity/issues/5315
-		"issue 5435":                      c.issue5435,                     // https://github.com/apptainer/singularity/issues/5435
-		"issue 5668":                      c.issue5668,                     // https://github.com/apptainer/singularity/issues/5435
-		"issue 5690":                      c.issue5690,                     // https://github.com/apptainer/singularity/issues/5690
+		"bad path":                        c.badPath,                              // try to build from a non existent path
+		"build encrypt with PEM file":     c.buildEncryptPemFile,                  // build encrypted images with certificate
+		"build encrypted with passphrase": c.buildEncryptPassphrase,               // build encrypted images with passphrase
+		"definition":                      c.buildDefinition,                      // builds from definition template
+		"from local image":                c.buildLocalImage,                      // build and image from an existing image
+		"from":                            c.buildFrom,                            // builds from definition file and URI
+		"multistage":                      c.buildMultiStageDefinition,            // multistage build from definition templates
+		"non-root build":                  c.nonRootBuild,                         // build sifs from non-root
+		"build and update sandbox":        c.buildUpdateSandbox,                   // build/update sandbox
+		"fingerprint check":               c.buildWithFingerprint,                 // definition file includes fingerprint check
+		"build with bind mount":           c.buildBindMount,                       // build image with bind mount
+		"library host":                    c.buildLibraryHost,                     // build image with hostname in library URI
+		"test with writable tmpfs":        c.testWritableTmpfs,                    // build image, using writable tmpfs in the test step
+		"test build system environment":   c.testBuildEnvironmentVariables,        // build image with build system environment variables set in definition
+		"test build under fakeroot modes": c.testContainerBuildUnderFakerootModes, // build image under different fakeroot modes
```

Comment:
Tests inside this function must be configured to run as sequential tests, not in parallel due to concerns above, you can use helper like https://github.com/apptainer/apptainer/blob/70cc654507d2f5a21c93195f93203ca600cb2a41/e2e/config/config.go#L822 and wrap the function to run tests sequentially

Reply:
thanks for the comment!
Addressed in this following commit
https://github.com/apptainer/apptainer/pull/607/commits/eeb2909f18d4b01efc8a05ca6a0c96949f059838